### ISSUE TITLE
Reuse `OC10` resource (code) as much as possible for `OCIS`

### ISSUE THREAD

**Issue by SagarGi** [owncloud/ocis#4509]
### Description

Recently we have been shifting some `spaces` related tests from core to ocis. During the process we have been duplicating code on ocis which is already in core. So we need to mitigate the code duplication and use the already available code (already in core) for new implementation on `ocis`. 

Note: Follow up this PR for reduction of code duplication.
https://github.com/owncloud/core/pull/40328
https://github.com/owncloud/ocis/pull/4492

#### NOTE:
Make possible code reduction related to the `features` that has been added during this issue:
https://github.com/owncloud/ocis/issues/4154#issuecomment-1206026045

**Comment by SagarGi** [comment]
TODO:
(Keep on adding features where you want to duplicate code if you find)

- [ ] lockspaces.feature (@SagarGi )

**Comment by amrita-shrestha** [comment]
@SagarGi can't we use core code for this also? https://github.com/owncloud/ocis/blob/b0fa917eae868b819f081290f6e0ab8159792d8b/tests/acceptance/features/bootstrap/SpacesContext.php#L520-L618

**Comment by SagarGi** [comment]
@amrita-shrestha i think we can but i think we cannot say its a duplicate code.

**Comment by amrita-shrestha** [comment]
> @amrita-shrestha i think we can but i think we cannot say its a duplicate code.

yeah but if we are reusing core code then why not properly implement it in overall ocis code?
I think it will be nice if you look over all ocis codes and check if there was pre-existing core code in ocis.

**Comment by SwikritiT** [comment]
> > @amrita-shrestha i think we can but i think we cannot say its a duplicate code.
> 
> yeah but if we are reusing core code then why not properly implement it in overall ocis code? I think it will be nice if you look over all ocis codes and check if there was pre-existing core code in ocis. This way we can provide consistency in ocis code.

I'll take a look what can be done for this

**Comment by SwikritiT** [comment]
I think I've reused as many tests as possible in https://github.com/owncloud/core/pull/40371 but if there are still other tests that can be reused that I might have left please take a look and mention here

@SagarGi @phil-davis @amrita-shrestha

**Comment by SagarGi** [comment]
@SwikritiT  can we also reuse for copy and move request? for (copySpaces.feature and moveSpaces.feature).

**Comment by SwikritiT** [comment]
> @SwikritiT can we also reuse for copy and move request? for (copySpaces.feature and moveSpaces.feature).

I'll take a look

**Comment by amrita-shrestha** [comment]
https://github.com/owncloud/ocis/blob/28c631163df87e9d09ca31aedb98df2495abacea/tests/acceptance/features/apiSpaces/changingFilesShare.feature#L17
https://github.com/owncloud/ocis/blob/28c631163df87e9d09ca31aedb98df2495abacea/tests/acceptance/features/bootstrap/SpacesContext.php#L1590-L1648

This step contain some core code maybe we can remove that too

**Comment by SwikritiT** [comment]
> https://github.com/owncloud/ocis/blob/28c631163df87e9d09ca31aedb98df2495abacea/tests/acceptance/features/apiSpaces/changingFilesShare.feature#L17
> 
> 
> https://github.com/owncloud/ocis/blob/28c631163df87e9d09ca31aedb98df2495abacea/tests/acceptance/features/bootstrap/SpacesContext.php#L1590-L1648
> 
> This step contain some core code maybe we can remove that too

We can reuse the code from the core for within space move or copy but form move in-between space, it's not possible through oc10 code so I don't think we should refactor these

**Comment by SagarGi** [comment]
@SwikritiT can you remove this method:
```
/**
	 * Send Propfind Request to Url
	 *
	 * @param  string $fullUrl
	 * @param  string $user
	 * @param  string $password
	 * @param  string $xRequestId
	 * @param  array  $headers
	 * @param mixed $body
	 *
	 * @return ResponseInterface
	 *
	 * @throws GuzzleException
	 */
	public function sendPropfindRequestToUrl(
		string $fullUrl,
		string $user,
		string $password,
		string $xRequestId = '',
		array $headers = [],
		$body = null
	): ResponseInterface {
		return HttpRequestHelper::sendRequest($fullUrl, $xRequestId, 'PROPFIND', $user, $password, $headers, $body);
	}
```


and make a direct request in the method: 
```
userListAllDeletedFilesInTrash```
May be we dont require the propfind method anymore since we are using it from `core`.

**Comment by SwikritiT** [comment]
> @SwikritiT can you remove this method:
> 
> ```
> /**
> 	 * Send Propfind Request to Url
> 	 *
> 	 * @param  string $fullUrl
> 	 * @param  string $user
> 	 * @param  string $password
> 	 * @param  string $xRequestId
> 	 * @param  array  $headers
> 	 * @param mixed $body
> 	 *
> 	 * @return ResponseInterface
> 	 *
> 	 * @throws GuzzleException
> 	 */
> 	public function sendPropfindRequestToUrl(
> 		string $fullUrl,
> 		string $user,
> 		string $password,
> 		string $xRequestId = '',
> 		array $headers = [],
> 		$body = null
> 	): ResponseInterface {
> 		return HttpRequestHelper::sendRequest($fullUrl, $xRequestId, 'PROPFIND', $user, $password, $headers, $body);
> 	}
> ```
> 
> and make a direct request in the method:
> 
> ```
> userListAllDeletedFilesInTrash```
> May be we dont require the propfind method anymore since we are using it from `core`.
> ```

Done in #4710

**Comment by SagarGi** [comment]
@SwikritiT I have scanned as much possible. I think we have reused lots of code from core. Can't really find out possible code to be reused. So may be this issue can be closed. @saw-jan @amrita-shrestha

**Comment by phil-davis** [comment]
I merged #4710 so this issue is done.
